### PR TITLE
SQLClientHelper should close connection when user-provided code fails

### LINKED ISSUE #205: SQLClientHelper.usingConnectionSingle has risk of the connection is not being closed correctly
public static <T> Single<T> usingConnectionSingle(SQLClient client, Function<SQLConnection, Single<T>> sourceSupplier) {
    return client.rxGetConnection().flatMap(conn -> {
      return sourceSupplier.apply(conn).doFinally(conn::close);
    });
  }
I think codes like "Single.flatMap(() -> sourceSupplier.apply(conn)).doFinally(conn::close);" should be better, as sourceSupplier.apply(conn)) may throw RuntimeException. Db connections are sensitive resources, so I think it's worth to ensure to close it.

The same problem to similar usingConnectionObservable etc

### PR DESCRIPTION
**Pull request by tsegismont**
Fixes #205

### CODE REVIEW

File: rx-java2/src/main/java/io/vertx/reactivex/ext/sql/SQLClientHelper.java [hunk 1]
```diff
@@ -226,7 +251,13 @@ public static Completable inTransactionCompletable(SQLClient client, Function<SQ
    */
   public static Completable usingConnectionCompletable(SQLClient client, Function<SQLConnection, Completable> sourceSupplier) {
     return client.rxGetConnection().flatMapCompletable(conn -> {
-      return sourceSupplier.apply(conn).doFinally(conn::close);
+      try {
+        return sourceSupplier.apply(conn).doFinally(conn::close);
+      } catch (Throwable t) {
+        Exceptions.throwIfFatal(t);
```
Comment (vietj) [inline rx-java2/src/main/java/io/vertx/reactivex/ext/sql/SQLClientHelper.java]: don't we want to throw after conn.close() ?

Comment (tsegismont) [reply rx-java2/src/main/java/io/vertx/reactivex/ext/sql/SQLClientHelper.java]: That's a general practice you can find in the RxJava codebase: if exception is fatal (e.g. OOM) you shouldn't do anything anyway (except logging it).

**Comment by tsegismont** [comment]
@vietj done

File: rx-java2/src/main/java/io/vertx/reactivex/ext/sql/SQLClientHelper.java (hunk 1 above)
Comment (vietj) [reply rx-java2/src/main/java/io/vertx/reactivex/ext/sql/SQLClientHelper.java]: ok, I was not aware of this

Comment (vietj) [reply rx-java2/src/main/java/io/vertx/reactivex/ext/sql/SQLClientHelper.java]: I mean the RxJava practice :-)

**Pull request merged**